proc_macros/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+use std::collections::HashMap;
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{
+    parse::{Parse, ParseStream},
+    parse_macro_input, Expr, ExprPath, Ident, Token,
+};
+
+struct BuilderArgs {
+    builder_name: ExprPath,
+    args: HashMap<Ident, Expr>,
+}
+
+impl Parse for BuilderArgs {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let builder_name: ExprPath = input.parse()?;
+        input.parse::<Token![,]>()?;
```

Comment:
I haven't really used this stuff. Do you know what that `Token` macro expands to?

Reply:
I did peek under the hood in doing some of the "rule macro" stuff (I think specifically I was trying to parse a literal `[` and didn't know if you could do eg `Token![[]` (you can't))

But didn't totally wrap my head around what it actually is, there's like an explicit mapping/lookup-table though to certain "pre-defined" individual/classes of tokens